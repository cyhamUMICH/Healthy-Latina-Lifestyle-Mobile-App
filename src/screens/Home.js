import React, { useState } from 'react';
import { Button } from 'react-native-elements';
import { Stylesheet, Text, View, Image, ScrollView, TouchableOpacity, FlatList, TextInput } from 'react-native';
import { styles } from '../styles/Styles'; 

const Home = (props) => {
  
  return (
    <View style={styles.app}>
        <Image 
        source={require('../../assets/logo-icon.png')} 
        style={styles.homepageLogo}
        />  
       <ScrollView
            horizontal
            showsHorizontalScrollIndicator={false}
            style={{ flex: 1 }}
          >
            <TouchableOpacity style={styles.homepageRoundButton} onPress={() => props.navigation.navigate("SetTimer")}>
              <Image source={require("../../assets/timer.png")} style={{ height: 32, width: 32 }}/>
            </TouchableOpacity>

            <TouchableOpacity style={styles.homepageRoundButton} onPress={() => props.navigation.navigate("MeditationList")}>
              <Image source={require("../../assets/meditation.png")} style={{ height: 43, width: 43 }}/>
            </TouchableOpacity>

            <TouchableOpacity style={styles.homepageRoundButton} onPress={() => props.navigation.navigate("ChallengeList")}>
              <Image source={require("../../assets/calendar-challenge.png")} style={{ height: 43, width: 43 }}/>
            </TouchableOpacity>

            <TouchableOpacity style={styles.homepageRoundButton} onPress={() => props.navigation.navigate("AddContent")}>
              <Image source={require("../../assets/addContent.png")} style={{ height: 43, width: 43 }}/>
            </TouchableOpacity>

            <TouchableOpacity style={styles.homepageRoundButton} onPress={() => props.navigation.navigate("YogaList")}>
            </TouchableOpacity>

            <TouchableOpacity style={styles.homepageRoundButton} onPress={() => props.navigation.navigate("Journal")}>
            </TouchableOpacity>

  
            </ScrollView>
      {/*<View style={{flex: 1, flexDirection: "column", justifyContent: "space-evenly"}}>
        < Button buttonStyle={styles.smallButton} titleStyle={styles.buttonText} title="Login" onPress={() => props.navigation.navigate("Login")} />
        <Button buttonStyle={styles.smallButton} titleStyle={styles.buttonText} title="Register" onPress={() => props.navigation.navigate("Register")} />
  <Button buttonStyle={styles.smallButton} titleStyle={styles.buttonText} title="Edit Account" onPress={() => props.navigation.navigate("EditAccount")} /> */}
      {/*} <Button buttonStyle={styles.smallButton} titleStyle={styles.buttonText} title="Meditation List" onPress={() => props.navigation.navigate("MeditationList")} />
         <Button title="Meditation" onPress={() => props.navigation.navigate("Meditation")} /> 
        <Button buttonStyle={styles.smallButton} titleStyle={styles.buttonText} title="Timer" onPress={() => props.navigation.navigate("Timer")} /> */}
</View> 
   
  );
};

export default Home;