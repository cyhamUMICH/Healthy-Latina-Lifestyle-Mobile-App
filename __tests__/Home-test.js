// Code from: https://github.com/byCedric/expo-guide-ci/blob/master/__tests__/App-test.js
//      and https://jestjs.io/docs/tutorial-react-native
import React from 'react';
import Home from '../src/screens/Home';
import renderer from 'react-test-renderer';

test('Renders Home With the Loading Screen', async () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders Home Without the Loading Screen', async () => {
  const tree = renderer.create(<Home skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});