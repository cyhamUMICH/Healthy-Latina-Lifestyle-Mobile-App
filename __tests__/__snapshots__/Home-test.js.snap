// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home With the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 70,
        "height": 100,
        "justifyContent": "center",
        "width": 80,
      }
    }
  />
  <RCTScrollView
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderRadius": 50,
            "flex": 1,
            "flexDirection": "row",
            "height": 66,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
            "width": 66,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 32,
              "width": 32,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderRadius": 50,
            "flex": 1,
            "flexDirection": "row",
            "height": 66,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
            "width": 66,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 43,
              "width": 43,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderRadius": 50,
            "flex": 1,
            "flexDirection": "row",
            "height": 66,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
            "width": 66,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 43,
              "width": 43,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderRadius": 50,
            "flex": 1,
            "flexDirection": "row",
            "height": 66,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
            "width": 66,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 43,
              "width": 43,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`Renders Home Without the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 70,
        "height": 100,
        "justifyContent": "center",
        "width": 80,
      }
    }
  />
  <RCTScrollView
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderRadius": 50,
            "flex": 1,
            "flexDirection": "row",
            "height": 66,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
            "width": 66,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 32,
              "width": 32,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderRadius": 50,
            "flex": 1,
            "flexDirection": "row",
            "height": 66,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
            "width": 66,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 43,
              "width": 43,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderRadius": 50,
            "flex": 1,
            "flexDirection": "row",
            "height": 66,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
            "width": 66,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 43,
              "width": 43,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00A89D",
            "borderRadius": 50,
            "flex": 1,
            "flexDirection": "row",
            "height": 66,
            "justifyContent": "center",
            "marginHorizontal": 10,
            "opacity": 1,
            "width": 66,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 43,
              "width": 43,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
